### PR TITLE
test: add sad paths for register with identifiers flow

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Add sad paths for register with identifiers flow

## Testing

Add response status guard for all interaction related APIs, We need do that after all tests is ready since a single flow will not cover all interaction API calls.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/integration-tests/src/tests/api/interaction/register-with-identifier/sad-path.test.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    // Reset
+    await updateSignInExperience({ signInMode: SignInMode.SignInAndRegister });
+  });
+
+  it('Should fail to register with username if related sign-up settings are not enabled', async () => {
+    await enableAllPasswordSignInMethods({
+      identifiers: [],
+      password: false,
+      verify: false,
+    });
```
Comment (simeng-li) [inline packages/integration-tests/src/tests/api/interaction/register-with-identifier/sad-path.test.ts:54]: before each?

Comment (xiaoyijun) [reply packages/integration-tests/src/tests/api/interaction/register-with-identifier/sad-path.test.ts]: Reorged & updated.

File: packages/integration-tests/src/tests/api/interaction/register-with-identifier/sad-path.test.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    await updateSignInExperience({ signInMode: SignInMode.SignInAndRegister });
+  });
+
+  describe('Should fail to register with identifiers if sign-up settings are not enabled', () => {
+    beforeAll(async () => {
+      // Disable all sign-up settings
+      await enableAllPasswordSignInMethods({
+        identifiers: [],
+        password: false,
+        verify: false,
```
Comment (simeng-li) [inline packages/integration-tests/src/tests/api/interaction/register-with-identifier/sad-path.test.ts:45]: same as default?

Comment (xiaoyijun) [reply packages/integration-tests/src/tests/api/interaction/register-with-identifier/sad-path.test.ts:45]: Upated with comments

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**